dashboard/test/models/user_test.rb

```diff
@@ -5455,4 +5455,24 @@ def hide_scripts_in_sections(section1, section2)
       end
     end
   end
+
+  describe 'new CAP columns data assigning before save' do
+    let(:user) {create(:student)}
+
+    it 'assigns "cap_state" with data from "child_account_compliance_state"' do
+      expected_cap_state = Policies::ChildAccount::ComplianceState::LOCKED_OUT
+
+      assert_changes -> {user.reload.cap_state}, from: nil, to: expected_cap_state do
+        user.update!(child_account_compliance_state: expected_cap_state)
+      end
+    end
+
+    it 'assigns "cap_state_date" with data from "child_account_compliance_state"' do
+      expected_cap_state_date = Time.now.change(usec: 0)
+
+      assert_changes -> {user.reload.cap_state_date}, from: nil, to: expected_cap_state_date do
+        user.update!(child_account_compliance_state_last_updated: expected_cap_state_date)
+      end
+    end
+  end
```

Comment:
These tests will also be removed:
`  # TODO(P20-1055): Remove in "CAP data migration Phase 2" once new CAP columns are populated`
See PR #59963